### PR TITLE
Fixes bug with ITL absolute height analysis, adds unit test, sets Znom = 12.992 mm

### PR DESCRIPTION
**Pull request by sethdigel**
The new iterative fitting scheme excludes outliers from the fits, and defines the mean and standard deviations of the filtered residuals.  For absolute height analysis (relevant only for ITL sensors) the reference plane is not fit, because it is defined in terms of the Znom absolute height.  The code was assuming that the mean and standard deviation of the filtered residuals were still being evaluated in this case.  The fix implemented here tests for that circumstance and evaluates those quantities before trying to use them. Ultimately the code should be refactored along the lines that Jim described to me.

I added a basic unit test to verify that the absolute height analysis is functioning for ITL sensors.  It uses a vendor data file and tests that the 0.000 and 1.000 quantile levels have not changed from what they were in a previous (now reference) analysis.

I updated the hardcoded definition of Znom for ITL sensors to the value for the production sensors, 12.992 mm.

### CODE REVIEW

File: tests/test_absoluteHeightITL.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+"""
+Executes an absolute height analysis for an ITL vendor data set, exercising
+code in set_ref_plane that formerly crashed on this data
+"""
+from __future__ import print_function
+import unittest
+#import numpy as np
```
Comment (jchiang87) [inline tests/test_absoluteHeightITL.py]: let's delete lines that aren't needed.

File: tests/test_absoluteHeightITL.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def tearDown(self):
+        pass
+
+    def test_quantile_table(self):
+        """
+        Verifies functionality of absolute height analysis for ITL
+        sensors by generating a quantile table and comparing it
+        with earlier results
+        """
+        infile = 'ITL_vendor_metrology_data.txt'
```
Comment (jchiang87) [inline tests/test_absoluteHeightITL.py]: The unit tests won't necessarily always be run from the `tests` subdirectory.  So full path to this file should be given, like this:
```
infile = os.path.join(os.environ['METROLOGYDATAANALYSISDIR'], 'tests', 'ITL_vendor_metrology_data.txt')
```
The `METROLOGYDATAANALYSISDIR` environment variable is set by the setup script generated by the `release/bin/install.py` script in the `lsst-camera-dh/release` package.

File: python/MetrologyData.py [hunk 3]
```diff
@@ -112,7 +111,21 @@ def set_ref_plane(self, plane_functor, zoffset=0, nsigma=5):
         pos, z = self.sensor.data()
         dz = z - plane_functor(pos) + zoffset
         self.resids = dz
+
         # Also define residuals with outliers removed (nsigma clipping)
+        if self.sensor.mean_filt == None:
```
Comment (jchiang87) [inline python/MetrologyData.py]: Standard python usage would instead be
```
if self.sensor.mean_filt is None:
```

**Review by jchiang87 — CHANGES_REQUESTED** [review]

File: python/MetrologyData.py (hunk 3 above)
Comment (sethdigel) [reply python/MetrologyData.py]: Thanks, Jim.  I've implemented this change and your recommendations below in the updates that I just committed.

**Review by jchiang87 — APPROVED** [review]
Looks good!  Please merge and resolve the JIRA issue, and I'll tag a release.

**Pull request merged**